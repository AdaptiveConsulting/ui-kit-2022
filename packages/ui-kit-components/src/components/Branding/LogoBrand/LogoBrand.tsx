import { useTheme } from '@mui/material';

import { darkLogoColor, lightLogoColor, LogoLarge } from '../branding';

interface Props {
  height?: number;
  width?: number;
}

const LogoBrand: React.FC<Props> = ({ height = 150, width = 148 }) => {
  const theme = useTheme();
  const color = theme.palette.mode === 'dark' ? lightLogoColor : darkLogoColor;

  return <LogoLarge sx={{ color: color, height: height, width: width }} />;
};

export default LogoBrand;
