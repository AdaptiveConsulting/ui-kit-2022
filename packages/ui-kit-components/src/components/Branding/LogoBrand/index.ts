import LogoBrand, { Props as LogoBrandProps } from './LogoBrand';
export { LogoBrand };
export type { LogoBrandProps };
