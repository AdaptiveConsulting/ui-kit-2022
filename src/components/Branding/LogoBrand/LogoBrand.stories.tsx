import { ComponentMeta, ComponentStory } from '@storybook/react';

import LogoBrand from './LogoBrand';

export default {
  title: 'Components/Branding',
  component: LogoBrand,
} as ComponentMeta<typeof LogoBrand>;

const Template: ComponentStory<typeof LogoBrand> = (args) => <LogoBrand {...args} />;

export const Brand = Template.bind({});

Brand.args = {};
