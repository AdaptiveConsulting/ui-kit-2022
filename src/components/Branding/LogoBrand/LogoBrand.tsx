import { SvgIcon } from '@mui/material';
import { useTheme } from '@mui/material/styles';

import { LogoLarge } from '../';

interface Props {
  height?: number;
  width?: number;
}

const LogoBrand: React.FC<Props> = ({ height = 150, width = 148 }) => {
  const theme = useTheme();
  const color = theme.palette.mode === 'dark' ? theme.palette.common.white : '#323232';

  return (
    <SvgIcon
      inheritViewBox
      component={LogoLarge}
      sx={{ color: color, height: height, width: width }}
    />
  );
};

export default LogoBrand;
