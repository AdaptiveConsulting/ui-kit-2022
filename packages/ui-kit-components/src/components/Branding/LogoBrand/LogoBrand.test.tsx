import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';

import LogoBrand from './LogoBrand';

jest.mock('../branding', () => ({
  __esModule: true,
  LogoLarge: ({ sx: { height, width } }: { sx: { height: number; width: number } }) => {
    return (
      <div data-testid={'logo-brand'}>
        <p>{height}</p>
        <p>{width}</p>
      </div>
    );
  },
}));

describe('LogoBrand component tests', () => {
  describe('Given a default LogoBrand', () => {
    beforeEach(() => {
      render(<LogoBrand />);
    });

    it('Then should be the same as the snapshot', () => {
      expect(screen.getByTestId('logo-brand')).toMatchSnapshot();
    });

    it('Then should render the LogoBrand component correctly', () => {
      expect(screen.getByTestId('logo-brand')).toBeInTheDocument();
    });

    it('Then should get the same width as default in the document text', () => {
      const element = screen.getByTestId('logo-brand');

      expect(element).toHaveTextContent('148');
    });

    it('Then should get the same height as default in the document text', () => {
      const element = screen.getByTestId('logo-brand');

      expect(element).toHaveTextContent('150');
    });
  });

  describe('Given a LogoBrand with customized size', () => {
    describe('When pass the custom props width or height', () => {
      beforeEach(() => {
        render(<LogoBrand width={50} height={100} />);
      });

      it('Then should get the same width as prop passed in the document text', () => {
        const element = screen.getByTestId('logo-brand');

        expect(element).toHaveTextContent('50');
      });

      it('Then should get the same height as prop passed in the document text', () => {
        const element = screen.getByTestId('logo-brand');

        expect(element).toHaveTextContent('100');
      });
    });
  });
});
