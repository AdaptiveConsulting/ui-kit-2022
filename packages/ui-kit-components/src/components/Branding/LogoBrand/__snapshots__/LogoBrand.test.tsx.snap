// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoBrand component tests Given a default LogoBrand Should be the same as the snapshots 1`] = `
<div
  data-testid="logo-brand"
>
  <p>
    150
  </p>
  <p>
    148
  </p>
</div>
`;
